### PR TITLE
Add ModulesClient tests for IotHub service client

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/src/ModulesClient.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ namespace Azure.Iot.Hub.Service
     public class ModulesClient
     {
         private const string ContinuationTokenHeader = "x-ms-continuation";
-        private const string HubModuleQuery = "select * from modules";
+        private const string HubModuleQuery = "select * from devices.modules";
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/ModulesClient.cs:21]: The service expected this format, not "select * from modules"

Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/ModulesClient.cs:21]: yeah I for everything they seem to have devices.xxx

**Review by vinagesh — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/ModulesClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
+using System.Threading.Tasks;
+using Azure.Iot.Hub.Service.Models;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using NUnit.Framework;
+
+namespace Azure.Iot.Hub.Service.Tests
+{
+    /// <summary>
+    /// Test all APIs of a DeviceClient.
```
Comment (bikamani) [inline sdk/iot/Azure.Iot.Hub.Service/tests/ModulesClientTests.cs:16]: Module client in summary :)

File: eng/Packages.Data.props [hunk 3]
```diff
@@ -164,5 +164,7 @@
     <PackageReference Update="Microsoft.Extensions.PlatformAbstractions" Version="1.1.0" />
 
     <PackageReference Update="Microsoft.Extensions.DependencyInjection" Version="2.1.0" />
+	
+    <PackageReference Update="Microsoft.Azure.Devices.Client" Version="1.26.0" />
```
Comment (bikamani) [inline eng/Packages.Data.props:168]: shouldn't this be 1.27.0?

Comment (timtay-microsoft) [reply eng/Packages.Data.props:168]: Sure thing

**Pull request merged**